File: Demo/Sources/ViewControllers/Notifications/Views/NotificationView.m
//
//  Globus
//

#import "NotificationView.h"

@interface NotificationView ()

@property(nonatomic, weak) IBOutlet GLBLabel* titleLabel;

@end

@implementation NotificationView

- (void)setTitle:(NSString*)title {
    _titleLabel.text = title;
}

- (NSString*)title {
    return _titleLabel.text;
}

#pragma mark - GLBNotificationViewProtocol

- (void)willShowNotificationView:(GLBNotificationView*)notificationView {
    NSLog(@"willShowNotificationView");
}

- (void)didShowNotificationView:(GLBNotificationView*)notificationView {
    NSLog(@"didShowNotificationView");
}

- (void)willHideNotificationView:(GLBNotificationView*)notificationView {
    NSLog(@"willHideNotificationView");
}

- (void)didHideNotificationView:(GLBNotificationView*)notificationView {
    NSLog(@"didHideNotificationView");
}

@end

File: Demo/Sources/ViewControllers/Notifications/Views/NotificationView.swift
//
//  Globus
//

import Globus

class NotificationView: GLBLoadedView, GLBNotificationViewProtocol {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var titleLabel: GLBLabel!
    
    // MARK - Public property
    
    public var title: String {
        set(value) {
            self.titleLabel.text = value
        }
        get {
            return self.titleLabel.text!;
        }
    }
    
    // MARK: - GLBNotificationViewProtocol
    
    internal func willShow(_ notificationView: GLBNotificationView) {
        print("willShowNotificationView")
    }
    
    internal func didShow(_ notificationView: GLBNotificationView) {
        print("didShowNotificationView")
    }
    
    internal func willHide(_ notificationView: GLBNotificationView) {
        print("willHideNotificationView")
    }
    
    internal func didHide(_ notificationView: GLBNotificationView) {
        print("didHideNotificationView")
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "NotificationView-Swift"
    }
    
}
